clients/python/tests/conftest.py

```diff
@@ -7,18 +7,71 @@
     ContextType,
     metadata_store_pb2,
 )
+from ml_metadata import errors
+from ml_metadata import metadata_store
+from ml_metadata.proto.metadata_store_pb2 import MetadataStoreClientConfig
 from model_registry.core import ModelRegistryAPIClient
 from model_registry.store.wrapper import MLMDStore
 from model_registry.types import ModelArtifact, ModelVersion, RegisteredModel
+import os
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.waiting_utils import wait_for_logs
+import time
 
 ProtoTypeType = Union[ArtifactType, ContextType]
 
 
-@pytest.fixture()
-def plain_wrapper() -> MLMDStore:
-    config = ConnectionConfig()
-    config.fake_database.SetInParent()
-    return MLMDStore(config)
+@pytest.fixture(scope='session')
+def plain_wrapper(request) -> MLMDStore:
+    print("Assuming this is Model Registry clients/python directory:", request.config.rootdir)
+    model_registry_root_dir = model_registry_root(request)
+    print("Assuming this is the Model Registry root directory:", model_registry_root_dir)
+    sqlite_db_file = model_registry_root_dir + '/test/config/ml-metadata/metadata.sqlite.db'
+    if os.path.exists(sqlite_db_file):
+        raise FileExistsError(f"The file {sqlite_db_file} already exists; make sure to cancel it before running these tests.")
+    container = DockerContainer("gcr.io/tfx-oss-public/ml_metadata_store_server:1.14.0")
+    container.with_exposed_ports(8080)
+    container.with_volume_mapping(model_registry_root_dir + '/test/config/ml-metadata/', '/tmp/shared', 'rw')
+    container.with_env('METADATA_STORE_SERVER_CONFIG_FILE', '/tmp/shared/conn_config.pb')
+    container.start()
+    wait_for_logs(container, "Server listening on")
+    os.system('docker container ls --format "table {{.ID}}\t{{.Names}}\t{{.Ports}}" -a')
+    print("waited for logs and port")
+    cfg = MetadataStoreClientConfig()
+    cfg.host = 'localhost'
+    cfg.port = int(container.get_exposed_port(8080))
+    print(cfg)
+
+    # this callback is needed in order to perform the container.stop()
+    # removing this callback might result in mlmd container shutting down before the tests had chance to fully run,
+    # and resulting in grpc connection resets.
+    def teardown():
+        os.system(f'rm {model_registry_root_dir}/test/config/ml-metadata/metadata.sqlite.db')
+        print("teardown")
+        container.stop()
+
+    request.addfinalizer(teardown)
+
+    time.sleep(3) # allowing some time for mlmd grpc to fully stabilize (is "spent" once per pytest session anyway)
+    store = metadata_store.MetadataStore(cfg)
+    wait_for_grpc(container, store)
+
+    return MLMDStore(cfg) 
+
+
+def model_registry_root(request):
+    return request.config.rootdir + '/../..'
```

Comment:
Per the previous comment, as `rootpath` is a `Path` object, it's possible to do:

```suggestion
def model_registry_root(request):
    return (request.config.rootpath / '../..').resolve()  # resolves to absolute path
```

you can use the same trick on other parts as well, printing this will also be fine :)